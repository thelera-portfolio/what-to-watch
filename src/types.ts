interface AuthData {
  login: string,
  password: string,
}

enum AuthorizationStatus {
  AUTH = `AUTH`,
  NO_AUTH = `NO_AUTH`,
};

interface Comment {
  comment: string,
  date: string,
  id: number,
  rating: number,
  user: {
    id: number,
    name: string,
  },
}

interface Film {
  backgroundColor: string,
  backgroundImage: string,
  description: string,
  director: string,
  genre: string,
  id: number,
  isFavourite: boolean,
  image: string,
  preview: string,
  previewVideoLink: string,
  ratingCount: number,
  ratingScore: number,
  runTime: number,
  starring: string[],
  title: string,
  videoLink: string,
  year: number,
}

enum FilmPageTab {
  OVERVIEW = `Overview`,
  DETAILS = `Details`,
  REVIEWS = `Reviews`,
};

enum Genre {
  ALL = `All genres`,
  ADVENTURE = `Adventure`,
  COMEDIES = `Comedy`,
  CRIME = `Crime`,
  DOCUMENTARY = `Documentary`,
  DRAMAS = `Drama`,
  FANTASY = `Fantasy`,
  HORROR = `Horror`,
  KIDS_AND_FAMILY = `Kid's and Family`,
  ROMANCE = `Romance`,
  SCI_FI = `Sci - Fi`,
  STORY = `Story`,
  THRILLERS = `Thriller`,
};

export {AuthData, AuthorizationStatus, Comment, Film, FilmPageTab, Genre};