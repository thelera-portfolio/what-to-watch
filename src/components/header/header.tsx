import * as React from "react";
import {AppRoute} from "../../utils/consts";
import {AuthorizationStatus} from "../../types";
import {connect} from "react-redux";
import {getAuthorizationStatus, getAvatarURL} from "../../reducer/user/selectors";
import {Link} from "react-router-dom";

interface Props {
  authorizationStatus: AuthorizationStatus;
  avatarImage: string;
  children: React.ReactNode;
  className: string;
  isMain: boolean;
  isLinkToMyList: boolean;
  isUserBlock: boolean;
}

const Header: React.FunctionComponent<Props> = (props: Props) => {
  const {authorizationStatus, avatarImage, children, className, isMain = false, isLinkToMyList, isUserBlock = false} = props;

  return (
    <header className={`page-header ${className}`}>
      <div className="logo">
        {
          isMain ?
            <a className="logo__link">
              <span className="logo__letter logo__letter--1">W</span>
              <span className="logo__letter logo__letter--2">T</span>
              <span className="logo__letter logo__letter--3">W</span>
            </a> :
            <Link to={!isMain && AppRoute.MAIN} className="logo__link">
              <span className="logo__letter logo__letter--1">W</span>
              <span className="logo__letter logo__letter--2">T</span>
              <span className="logo__letter logo__letter--3">W</span>
            </Link>
        }
      </div>

      {children}

      {
        isUserBlock ?
          <div className="user-block">
            <Link to={isLinkToMyList && AppRoute.MY_LIST} className="user-block__link">
              {authorizationStatus === AuthorizationStatus.NO_AUTH && `Sign In`}
              {authorizationStatus === AuthorizationStatus.AUTH &&
              <div className="user-block__avatar">
                <img src={avatarImage} alt="User avatar" width="63" height="63" />
              </div>}
            </Link>
          </div> :
          <div className="user-block">
            <div className="user-block__avatar">
              {avatarImage && <img src={avatarImage} alt="User avatar" width="63" height="63" />}
            </div>
          </div>
      }
    </header>
  );
};

const mapStateToProps = (state) => ({
  authorizationStatus: getAuthorizationStatus(state),
  avatarImage: getAvatarURL(state),
});

export {Header};
export default connect(mapStateToProps)(Header);
