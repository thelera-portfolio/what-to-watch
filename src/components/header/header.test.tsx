import * as React from "react";
import * as renderer from "react-test-renderer";
import {AuthorizationStatus} from "../../types";
import {BrowserRouter} from "react-router-dom";
import {Header} from "./header";

it(`Header is rendered correctly`, () => {
  const tree = renderer.create(
      <BrowserRouter>
        <Header
          authorizationStatus={AuthorizationStatus.NO_AUTH}
          avatarImage={`image`}
          className={`class`}
          isMain={false}
          isLinkToMyList={true}
          isUserBlock={true}
        >
          <div></div>
        </Header>
      </BrowserRouter>).toJSON();

  expect(tree).toMatchSnapshot();
});
