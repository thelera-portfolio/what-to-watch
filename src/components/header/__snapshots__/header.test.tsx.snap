// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header is rendered correctly 1`] = `
<header
  className="page-header class"
>
  <div
    className="logo"
  >
    <a
      className="logo__link"
      href="/"
      onClick={[Function]}
    >
      <span
        className="logo__letter logo__letter--1"
      >
        W
      </span>
      <span
        className="logo__letter logo__letter--2"
      >
        T
      </span>
      <span
        className="logo__letter logo__letter--3"
      >
        W
      </span>
    </a>
  </div>
  <div />
  <div
    className="user-block"
  >
    <a
      className="user-block__link"
      href="/mylist"
      onClick={[Function]}
    >
      Sign In
    </a>
  </div>
</header>
`;
